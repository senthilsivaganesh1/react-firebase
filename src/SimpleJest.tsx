import { useState } from "react";

export default function SimpleJest() {
    const [name, setName] = useState<string>("");
    return (
        <>
            <input type="text" onChange={(e)=>setName(e.target.value)} data-testid="input" aria-label="input"/>
            <label data-testid="output">{`hello ${name}`}</label>
        </>
    )
}