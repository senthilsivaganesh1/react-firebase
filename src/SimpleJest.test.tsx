import { fireEvent, render } from "@testing-library/react"
import SimpleJest from "./SimpleJest"

describe("SimpleJest",()=>{
    it("is app loading", ()=> {
        const {baseElement} = render(<SimpleJest/>)
        expect(baseElement).toBeInTheDocument();
    })    
    it("Check if onChange event is working", ()=> {
        const {getByTestId} = render(<SimpleJest/>)
        const input = getByTestId("input");
        fireEvent.change(input, {target:{value:"geeks"}});
        expect(input).toHaveValue("geeks");
        const output = getByTestId("output");
        expect(output.innerHTML).toBe("hello geeks");
    })
    it("match snapshot testing", ()=> {
        const {baseElement} = render(<SimpleJest/>)
        expect(baseElement).toMatchSnapshot();
    })

})
