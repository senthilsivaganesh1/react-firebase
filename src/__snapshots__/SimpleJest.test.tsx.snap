// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleJest match snapshot testing 1`] = `
<body>
  <div>
    <input
      aria-label="input"
      data-testid="input"
      type="text"
    />
    <label
      data-testid="output"
    >
      hello 
    </label>
  </div>
</body>
`;
